deepset_cloud_sdk/workflows/sdk.py

```diff
@@ -0,0 +1,87 @@
+"""Deepset AI platform SDK main class."""
+import structlog
+
+from deepset_cloud_sdk.workflows.pipeline_client.pipeline_service import (
+    _enable_import_into_deepset,
+)
+
+logger = structlog.get_logger(__name__)
+
+
+class DeepsetSDK:
+    """Main class for Deepset Cloud SDK functionality.
+
+    This class provides a centralized way to initialize and manage SDK features.
+
+    Example for importing an index or pipeline to deepset AI platform:
+        ```python
+        from deepset_cloud_sdk.workflows import DeepsetSDK
+        from haystack import Pipeline
+
+        # Initialize the SDK to enable using importing functionality
+        sdk = DeepsetSDK()
+        sdk.init()
+
+        # Configure your pipeline
+        pipeline = Pipeline()
+
+        # Configure import
+        # if importing an index, use IndexConfig
+        config = IndexConfig(
+            name="my-index",
+            inputs=IndexInputs(files=["file_type_router.sources"]),
+        )
+
+        # if importing a pipeline, use PipelineConfig
+        config = PipelineConfig(
+            name="my-pipeline",
+            inputs=PipelineInputs(
+                query=["prompt_builder.query"],
+                filters=["bm25_retriever.filters", "embedding_retriever.filters"],
+            ),
+            outputs=PipelineOutputs(
+                answers="answers_builder.answers",
+                documents="ranker.documents",
+            ),
+        )
+
+        # sync execution
+        pipeline.import_into_deepset(config)
+
+        # async execution
+        await pipeline.import_into_deepset_async(config)
+        ```
+    """
+
+    def __init__(self) -> None:
+        """Initialize the DeepsetSDK instance."""
+        self._initialized = False
+
+    @property
+    def is_initialized(self) -> bool:
```

Comment:
what do we need that for?

Reply:
Could be used to check if the sdk is initialized, i.e. methods were added, but not necessarily needed. Removed.